src/event-listeners/index.js

```diff
@@ -108,14 +112,60 @@ export default function setupEventListeners(store, canvas) {
   // kill performance
   canvas.addEventListener('mousemove', event => {
     const state = store.getState()
+    const { goals, edges, ui: { viewport: { translate, scale }, screensize: { width } }} = state
     if (state.ui.mouse.mousedown) {
-      store.dispatch(changeTranslate(event.movementX, event.movementY))
-      return
+     
+      if(event.shiftKey){
+        if(boolean===false){
+          x=event.clientX
+          y=event.clientY
+           convertedClick = coordsPageToCanvas({
+            x: event.clientX,
+            y: event.clientY
+        }, translate, scale)
+          boolean=true
+        }
+        
+         let lienzo = canvas.getContext("2d")
+         store.dispatch(unhoverGoal())
+         const convertedIni = coordsPageToCanvas({
+          x: event.clientX,
+          y: event.clientY
+      }, translate, scale)
+         let xd=convertedIni.x-convertedClick.x
+         let yd=convertedIni.y-convertedClick.y
+
+         lienzo.beginPath()
+        
+        lienzo.moveTo(convertedClick.x,convertedClick.y)
+        lienzo.lineTo(convertedClick.x+xd,convertedClick.y)
+        lienzo.lineTo(convertedClick.x+xd,convertedClick.y+yd)
+        lienzo.lineTo(convertedClick.x,convertedClick.y+yd)
+        lienzo.lineTo(convertedClick.x,convertedClick.y)
+
+        lienzo.closePath()
+        lienzo.stroke()
+        
+        goalAddresses = checkForGoalAtCoordinatesInBox(canvas.getContext('2d'), translate, scale, width, goals, edges, event.clientX, event.clientY,x,y)
+      }else{ store.dispatch(changeTranslate(event.movementX, event.movementY))}
+
+    }else{
+      if(goalAddresses!==null&&boolean){
+        boolean=false;
+        console.log(goalAddresses)
+        goalAddresses.forEach(value=>(store.dispatch(selectGoal(value))))
+        goalAddresses=null
+        x=0
+        y=0
+        state.ui.selection.selectedGoals=[]
```

Comment:
we can't modify state that comes from redux directly, so we have to remove this line ^